### ISSUE TITLE
TPL Updates for version 0.98.8

### ISSUE THREAD

**Issue by jd-moulton** [amanzi/amanzi#792]
Our TPL updates (#742) slipped behind the update of Trilinos and Hypre (which became TPLs 0.98.7).  We need to recapture the updates of several libraries that were integrated in the old TPLs 0.98.7 branch.   This new ticket (and branch cut from current master) will capture that work.  

We need to aggregate updates for TPLs coming in from PRs to avoid users having to rebuild the tpls too often. We may integrate other minor updates to some libraries as well. The current PRs are:

- [x] - update ExprTk package (https://github.com/amanzi/amanzi/pull/734)
- [x] - patch netcdf-fortran cmake (https://github.com/amanzi/amanzi/pull/735)
- [x] - fix confusing naming (https://github.com/amanzi/amanzi/pull/736)

In addition, to prepare for the work on gas diffusion we need to update geochemistry related libraries

- [x]  alquimia v1.1.0
- [x]  PetSc v3.20.0
- [x]  Sowing (1.1.26-p8) that goes with PetSc
- [x]  PFLOTRAN v5.0.0

And we need to restore the ability to build either CrunchFlow or PFLOTRAN, but not require both.

Other updates are possible but may wait (e.g., boost).

**Comment by jd-moulton** [comment]
**Note:** these updates will fail CI until I reach a point where it is worth updating the Docker TPLs image to match the new version.

**Comment by ecoon** [comment]
This fails for me:  

CMake Error at tools/cmake/AmanziVersion.cmake:123 (string):
  string sub-command REGEX, mode MATCH needs at least 5 arguments total to
  command.
Call Stack (most recent call first):
  CMakeLists.txt:78 (include)

line 123 is:

`  string(REGEX MATCH "\\[(.*)\\]" _ ${AMANZI_GIT_PARENT_BRANCH_MSG})`

I'm not sure what you're trying to do here

**Comment by ecoon** [comment]
For now I just commented out this line and the following line, and moved on....

**Comment by ecoon** [comment]
Also this does not work for me:

AmanziTPL.cmake:26

```string(REGEX REPLACE ".*AMANZI_TPLS_VERSION_PATCH ([0-9]+)([a-z]+).*" "\\1\\2" AMANZI_TPLS_VERSION_PATCH_SRC "${line}")```

When the patch has no letters at the end, it does not match and I get:

```
-- >>>>>>>> AmanziTPL.cmake
-- 	 >>>>>> set(AMANZI_TPLS_VERSION_PATCH 9)
-- 	 >>>>>> Amanzi Source Path:       /Users/Shared/ornldev/code/ats/ats/repos/david-amanzi-tpls-0.98.8
-- 	 >>>>>  Amanzi Required Version:  0.98.set(AMANZI_TPLS_VERSION_PATCH 9)
-- 	 >>>>>  Amanzi TPL Version:       0.98.9
```

**Comment by jd-moulton** [comment]
The messaging about required and actual TPL Version is for informational purposes because developers may have a reason to use a different version.  But for users the ideas was these would line up and help us help them. 

The REGEX foo needs work for the TPL version with appended letters.  I removed it from the Docker stuff, had forgotten it was here too since the builds don't fail, it's just the information is inaccurate/broken.

I pushed an update that removes it for now.

**Comment by jd-moulton** [comment]
> This fails for me:
> 
> CMake Error at tools/cmake/AmanziVersion.cmake:123 (string): string sub-command REGEX, mode MATCH needs at least 5 arguments total to command. Call Stack (most recent call first): CMakeLists.txt:78 (include)
> 
> line 123 is:
> 
> ` string(REGEX MATCH "\\[(.*)\\]" _ ${AMANZI_GIT_PARENT_BRANCH_MSG})`
> 
> I'm not sure what you're trying to do here

The error message indicates that the variable ${AMANZI_GIT_PARENT_BRANCH_MSG} is empty so it thinks it doesn't have enough arguments.  As the name implies this code is trying to glean information about how this branch is related to master so it can create a more informative version tag.   I'm not sure what your branch info looks like, can you post the output of "git show-branch", and we can figure it out.  Also, seems like there should be a way to pass an empty string if the variable is undefined or empty, then at least it wouldn't error.

**Comment by ecoon** [comment]
`git show-branch` gives:

`[david/amanzi-tpls-0.98.8] Merge branch 'master' into david/amanzi-tpls-0.98.8`

**Comment by ecoon** [comment]
Where is AMANZI_GIT_PARENT_BRANCH originally defined?  It is empty, but grepping through cmake files shows no step that makes the env variable, and there is no such variable in my amanzi-tpl-config.cmake, and no reference to it in bootstrap, so I'm not sure where it gets defined initially.

**Comment by jd-moulton** [comment]
Did you clone just the branch?  Maybe that's why and I'll need to punt in that case instead and make sure the variable is not empty (could be "unknown" or something.  Here's what mine shows, you can see the "+" by the parent branch.

```
* [david/amanzi-tpls-0.98.8] Merge branch 'david/amanzi-tpls-0.98.8' of https://github.com/amanzi/amanzi into david/amanzi-tpls-0.98.8
 ! [master] Added Schiffman test with linear load and non-trivial Biot coefficient.
--
-  [david/amanzi-tpls-0.98.8] Merge branch 'david/amanzi-tpls-0.98.8' of https://github.com/amanzi/amanzi into david/amanzi-tpls-0.98.8
*  [david/amanzi-tpls-0.98.8^2] Removed other appearance of failed REGEX foo for TPLs with appended letters (refs #792).
*  [david/amanzi-tpls-0.98.8^] Updated deployment script to build TPLs from this branch (refs #792).
*+ [master] Added Schiffman test with linear load and non-trivial Biot coefficient.

**Comment by ecoon** [comment]
I did clone the branch directly.

**Comment by ecoon** [comment]
@jd-moulton  can this be closed now that we are already using 0.98.9?
